packages/bioformats/files/components/xsd-fu/templates-cpp/OMEXMLModelEnum.template

```diff
@@ -0,0 +1,554 @@
+          string_map_type::const_iterator i = string_map.find(name);
+          string_map_type::const_iterator li = lowercase_string_map.end();
+          if (strict == false && i == string_map.end())
+            {
+              // Trim leading and trailing whitespace and lowercase.
+              // NOTE: This lowercasing operation only works with 7-bit ASCII, not UTF-8.
+              // NOTE: To make this work properly with Unicode, look at ICU.
+              std::string lname(name);
+              trim(lname);
+              std::transform(lname.begin(), lname.end(), lname.begin(), std::ptr_fun(::tolower));
+              li = lowercase_string_map.find(lname);
+{% if 'Other' in klass.possibleValues %}\
+              if (li == lowercase_string_map.end())
+                {
+                  ome::common::Logger logger = ome::common::createLogger("${klass.langType}");
+                  format fmt("Unsupported %1% value of ‘%2%’ will be stored as “Other”");
+                  fmt % "${klass.langType}" % name;
+                  BOOST_LOG_SEV(logger, ome::logging::trivial::warning) << fmt.str();
+                  i = string_map.find("Other");
+                }
+              else
+                {
+                  ome::common::Logger logger = ome::common::createLogger("${klass.langType}");
+                  format fmt("Unsupported %1% value of ‘%2%’ will be stored as “%3%”");
+                  fmt % "${klass.langType}" % name % ${klass.langType}(li->second);
+                  BOOST_LOG_SEV(logger, ome::logging::trivial::warning) << fmt.str();
+                }
+{% end %}\
+            }
+          if (i != string_map.end())
+	    {
+	      this->value = i->second;
+	    }
+	  else if (li != lowercase_string_map.end())
+	    {
+	      this->value = li->second;
+	    }
+	  else
+            {
+              format fmt("‘%1%’ not a supported value of %2%");
+              fmt % name % "${klass.langType}";
+              throw EnumerationException(fmt.str());
+            }
+
+          const value_map_type& value_map(values());
+          value_map_type::const_iterator i2 = value_map.find(value);
+          if (i2 == value_map.end())
+            {
+              format fmt("‘%1%’ not a supported value of %2%");
+              fmt % value % "${klass.langType}";
+              throw EnumerationException(fmt.str());
+            }
+          this->name = &i2->second;
+        }
+
+        ${klass.langType}::${klass.langType} (const ${klass.langType}& original):
+          value(original.value),
+          name(original.name)
+        {
+        }
+
+        const ${klass.langType}::string_map_type&
+        ${klass.langType}::strings()
+        {
+	  const smap init_string_values[] =
+	    {
+{% for value in klass.possibleValues %}\
+{% if value == klass.possibleValues[-1] %}\
+              smap("${value}", ${klass.langTypeNS}::${enum_value_name(value, False).upper()}),
```

Comment:
It should be to avoid a comma at the end of the list which makes the compiler upset, but it looks like that isn't the case here.